src/test/java/com/waverleysoftware/AccessCodeTests.java

```diff
@@ -1,32 +1,36 @@
 package com.waverleysoftware;
 
-import org.testng.annotations.AfterClass;
-import org.testng.annotations.BeforeClass;
+import com.waverleysoftware.data.DataSuppliers;
+import com.waverleysoftware.data.Name;
+import com.waverleysoftware.model.Panel;
+import lombok.extern.slf4j.Slf4j;
 import org.testng.annotations.Test;
 
-public class AccessCodeTests extends BaseTest {
-    @BeforeClass
-    public void setUp() {
-        System.out.println("Setting up on child level");
-    }
+import static java.util.Arrays.asList;
 
-    @Test(priority = 1)
-    public void accessCodeShouldBeCreated() {
-        System.out.println("Access code created");
+/**
+ * A simple model class for access code testing.
+ */
+@Slf4j
+public class AccessCodeTests extends BaseTest {
+    @Test(dataProvider = "getPanelData", dataProviderClass = DataSuppliers.class)
+    public void accessCodeShouldBeAdded(final Panel... panels) {
+        final String conditionForType = "home";
+        SOFTASSERT.assertThat(panels[0].getType()).as("Panel Name").isEqualTo(conditionForType);
     }
 
-    @Test(priority = 2)
-    public void accessCodeShouldBeModified() {
-        System.out.println("Access code modified");
+    @Name(name = "example", type = "code")
+    public String getName() {
+        return "someName";
     }
 
-    @Test(priority = 3)
-    public void accessCodeShouldBeDeleted() {
-        System.out.println("Access code deleted");
+    @Test(dataProvider = "getPanelData", dataProviderClass = DataSuppliers.class)
+    public void accessCodeShouldBeCreatedWithSomeDates(final Panel... panels) {
+        SOFTASSERT.assertThat(asList(panels).get(0).getName()).as("panel name").isEqualTo("lyn");
```

Comment:
if you're always using first element, why don't you return just a single object from data supplier?